### ISSUE TITLE
Checkboxes (aka Tickboxes) missing required aria-label attribute with unique ID

### ISSUE THREAD

**Issue by katannshaw** [olifolkerd/tabulator#3635]
**Describe the bug**
Checkboxes used for the tickbox feature are missing a required label element or `aria-label` attribute per the following WCAG guidelines: <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html">1.3.1 Info and Relationships</a> and <a href="https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html">4.1.2 Name, Role, Value</a>

**Tabulator Info**
Branch 5.1

**Working Example**
https://codepen.io/katannshaw/pen/eYeQKEy

**To Reproduce**
1. Go to https://codepen.io/katannshaw/pen/eYeQKEy
2. Run an accessibility check using any free tool like [axe Dev Tools Chrome extension](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US&_ga=2.93270173.1290087488.1646232283-1243334307.1639763157), [HTML Code Sniffer bookmarklet](https://squizlabs.github.io/HTML_CodeSniffer/), or what I used [ARC Toolkit Chrome extension](https://chrome.google.com/webstore/detail/arc-toolkit/chdkkkccnlfncngelccgbgfmjebmkmce?hl=en-US)
3. Verify that you receive the following accessibility errors

> 4.1.2 Name, Role, Value
This checkboxinput element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.

> 1.3.1 Info and Relationships
This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.

**Expected behavior**
Because the labels need to be hidden, the best technique would be to add an `aria-label` attribute with a unique ID to each checkbox like this:
```
<input type="checkbox" aria-label="tabulator-id-1">
<input type="checkbox" aria-label="tabulator-id-2">
<input type="checkbox" aria-label="tabulator-id-3">
```

...where 1, 2, and 3 would be a unique ID from the database, not the order of the rows in the table since they can be moved.

**Screenshots**
![Screen Shot 2022-03-02 at 8 49 51 AM](https://user-images.githubusercontent.com/13474823/156385425-da28eac1-19d5-47a5-ab7d-bf5f471a0144.png)

**Desktop (please complete the following information):**
 - OS: macOS Big Sur v11.2.3
 - Browser: Chrome
 - Version 98.0.4758.102

**Smartphone (please complete the following information):**
 - Device: iPhoneX
 - OS: iOS15.3.1
 - Browser Safari
 - Version 14.0.3

**Additional context**
The WCAG guidelines mentioned are in both [WCAG 2.0](https://www.w3.org/TR/WCAG20/) and [WCAG 2.1](https://www.w3.org/TR/WCAG21/)

**Comment by olifolkerd** [comment]
Hey @katannshaw 

I will add an aria label in the next patch release. though i see no need for a unique name, they all perform the same function, of select row, which would be the equivalent label if it is set, so for now i will default them all to have that label.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @katannshaw 

I have pushed a fix for this to the master branch and will include it in this weekends patch release.

Cheers

Oli :)

**Comment by katannshaw** [comment]
> Hey @katannshaw
> 
> I will add an aria label in the next patch release. though i see no need for a unique name, they all perform the same function, of select row, which would be the equivalent label if it is set, so for now i will default them all to have that label.
> 
> Given the point of the label is to provide context and often actual vocalisations to screen readers giving it an arbitrary unique value would be unhelpful
> 
> Cheers
> 
> Oli :)

Thanks for pushing a fix, that was fast!

However, not having a unique ID for each checkbox element will still fail accessibility testing with [WCAG F77: Failure of Success Criterion 4.1.1 due to duplicate values of type ID](https://www.w3.org/TR/WCAG20-TECHS/F77.html).

This correct technique to use can seen in the [MDN Web Docs for <input type="checkbox">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox) and [MDN Web Docs aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) It's only in radiobutton groups that they can pass accessibility testing with having a unique ID while sharing the same name attribute, as seen in the [MDN Web Docs for <input type="radio">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)

So I recommend a fix in this case where each checkbox has its own `aria-label` with a unique ID, which acts like a visually label using `ARIA` that's read aloud to screen reader users. You can use a system ID so that you don't have to worry about its value changing.

```
<input type="checkbox" aria-label="tabulator-id-123"><!-- Has system ID 123 -->
<input type="checkbox" aria-label="tabulator-id-456"><!-- Has system ID 456 -->
<input type="checkbox" aria-label="tabulator-id-789"><!-- Has system ID 789 -->
```

Hope it helps. Thanks.

Kat
